infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingApplier.java

```diff
@@ -124,8 +127,30 @@ public void apply(
 
       Collection<CommandImpl> pluginRelatedCommands = commandsResolver.resolve(chePlugin);
 
+      Optional<ComponentImpl> pluginRelatedComponentOpt =
+          k8sEnv
+              .getDevfile()
+              .getComponents()
+              .stream()
+              .filter(c -> chePlugin.getId().equals(c.getId()))
+              .findAny();
+      if (!pluginRelatedComponentOpt.isPresent()) {
+        throw new InfrastructureException(
+            "The downloaded plugin configuration does not have the "
```

Comment:
Maybe mention the involved plugin IDs (the chePlugin's and the ones from the devfile) for better debuggability?

Reply:
Added such info into message and test case to make sure that it's well-formatted